install_config/persistent_storage/persistent_storage_flex_volume.adoc

```diff
@@ -0,0 +1,304 @@
+Other parameters are specific to each operation. Most of the operations expect
+JSON string as a parameter. This parameter is literally a whole JSON string,
+not name of a file with the JSON! It contains all `flexVolume.options`, some
+options from `flexVolume` prefixed by `kubernetes.io/` and content of the
+referenced secret (if specified) prefixed by `kubernetes.io/secret/`.
+
+.Flex Driver JSON input
+====
+[source,json]
+----
+{
+	"fooServer": "192.168.0.1:1234", <1>
+        "fooVolumeName": "bar",
+	"kubernetes.io/fsType": "ext4", <2>
+	"kubernetes.io/readwrite": "ro", <3>
+	"kubernetes.io/secret/<key name>": "<key value>", <4>
+	"kubernetes.io/secret/<another key name>": "<another key value>",
+}
+----
+<1> All options from `flexVolume.options`.
+<2> Value of `flexVolume.fsType`.
+<3> `ro`/`rw` based on `flexVolume.readOnly`.
+<4> All keys and their values from the secret referenced by `flexVolume.secretRef`.
+====
+
+OpenShift expects a JSON on standard output of the driver. When not specified
+otherwise, the output describes result of the operation.
+
+.Flex Driver default output
+====
+[source,json]
+----
+{
+	"status": "<Success/Failure/Not supported>",
+	"message": "<Reason for success/failure>"
+}
+----
+====
+
+Exit code of the driver should be 0 for success and 1 for error.
+
+Operations should be idempotent, i.e. attaching of already attached volume or
+mounting of already mounted volume should result in operation success.
+
+Flex Volume driver can work in two modes, with attach/detach operation or
+without it. Attach/detach operation is used by OpenShift master
+to attach a volume to a node and to detach it from a node. This
+is useful when a node becomes unresponsive from any reason - master then can
+kill all pods on the node, detach all volumes from it and attach the volumes to
+another nodes to resume the applications, while the original node is still not
+reachable.
+
+Not every storage backend supports master-initiated detachment of a volume from
+another machine, please consider carefully what kind of Flex Volume driver
+are you going to write.
+
+==== Flex Volume drivers with master-initiated attach/detach
+
+Flex Volume driver that supports master controlled attach/detach must implement
+these operations:
+
+`init`::
+* Initializes the driver. It is called during initialization of masters and nodes.
+* Arguments: none
+* Executed on: master, node
+* Expected output: default JSON
+
+`getvolumename`::
+* Returns unique name of the volume. This name must be consistent among all
+masters and nodes and will be used in subsequent `detach` call as
+`<volume-name>`. Any `'/'` character in the `<volume-name>` are automatically
+replaced by `'~'`.
+* Arguments: `<json>`
+* Executed on: master, node
+* Expected output: default JSON + `volumeName`:
++
+[source,json]
+----
+{
+	"status": "Success",
+	"message": "",
+	"volumeName": "foo-volume-bar" <1>
+}
+----
+<1> Unique name of the volume in storage backend foo.
```

Comment:
Well, the storage back-end may provide many volumes. And this is the unique identifier of the volume on the back-end.